Fix scripts for unit and e2e testing

gsambrotta opened the pull request:
This PR simply improve the naming of the test script and change the arguments to reflect our current testing file structure

lukestanley commented on package.json:
```diff
@@ -16,10 +16,10 @@
+    "test:serve_and_retest_on_change": "xvfb-maybe start-server-and-test serve:dev http://localhost:4200 'npm run test:retest_on_change'",
+    "test:retest_on_change": "xvfb-maybe nodemon -e js,jsx,ts,tsx,css,scss --exec 'ng test --watch=false; cypress open --config watchForFileChanges=true'",
+    "test:autorun": "xvfb-maybe ng test --watch=false; xvfb-maybe cypress run --headed",
+    "Compodoc": "npx cypress install; xvfb-maybe start-server-and-test serve:dev http://localhost:4200 'ng test --watch=false; cypress run'",
```
What's Compodoc?

gsambrotta replied:
:upside_down_face: This is a really good question @lukestanley! I have no idea :D 
Some time as passed and I don't know why I put that name. Well, obviously is not a good name :sweat_smile:

lukestanley replied:
Clearly, it has little to do with Compodoc and should be fixed! @gsambrotta

CodeKrakken reviewed (approved):
No problems here. Naming is clearer and functionality of `test:serve_and_retest_on_change` is improved.

gsambrotta commented:
> I think the README needs updating if command names are changed. It's not obvious to me what the new commands do. Changing less at a time may be better?

Which new commands are not clear to you? Is about the naming or about what the command does in itself?

Honestly, I didn't change much. 
I changed `e2e` to `test` and I fixed the scripts with working arguments, But the commands do the same as they were doing before.
@lukestanley

gsambrotta commented:
@lukestanley ready to be reviewed again. 
I updated the Readme too.

lukestanley reviewed (approved):
This is clearer now, nice work! Maybe I'll make a Mermaid decision flow chart with the choices for launching the app! ;D
